libs/entitlement/asm-provider/src/main/java/org/elasticsearch/entitlement/instrumentation/impl/InstrumenterImpl.java

```diff
@@ -273,9 +288,9 @@ private void invokeInstrumentationMethod() {
     protected void pushEntitlementChecker(MethodVisitor mv) {
         mv.visitMethodInsn(
             INVOKESTATIC,
-            "org/elasticsearch/entitlement/bridge/EntitlementCheckerHandle",
+            handleClass,
             "instance",
-            "()Lorg/elasticsearch/entitlement/bridge/EntitlementChecker;",
+            "()L" + checkerClass + ";",
```

Comment:
Nit: using `Type.getObjectType(checkerClass).getDescriptor()` is probably more robust (and could be moved to the ctor too)